Sources/NIO/SocketChannel.swift

```diff
@@ -53,8 +53,17 @@ final class SocketChannel: BaseSocketChannel<Socket> {
         return super.isOpen
     }
 
-    init(eventLoop: SelectableEventLoop, protocolFamily: Int32) throws {
+    convenience init(eventLoop: SelectableEventLoop, protocolFamily: Int32) throws {
         let socket = try Socket(protocolFamily: protocolFamily, type: Posix.SOCK_STREAM, setNonBlocking: true)
+        try self.init(eventLoop: eventLoop, socket: socket)
+    }
+
+    convenience init(eventLoop: SelectableEventLoop, descriptor: Int32) throws {
+        let socket = try Socket(descriptor: descriptor, setNonBlocking: true)
+        try self.init(eventLoop: eventLoop, socket: socket)
+    }
```

Comment:
Can we have this for the `BaseSocketChannel` instead? As noted, it'd be useful for this to generally be available.